the-collab-lab/tcl-34-smart-shopping-list
# Ordering products by time and name

## Description
 ```firebaseUtils```:
 * First we transformed the data to a new key value to the object, and this object contains the string value that is going to be used for both: CSS colors and aria-label. The ladder works with a function called timeframeLabel which is stored in ```timeFrame.js```
 * After this, we sort the products to order them by the same day in ```daysUntilNextPurchase```and if they are the same these are ordered alphabetically. Otherwise, we ordered them by daysUntilNextPurchase in ascending order.
 * In timeFrame we delete the ```timeFrame```  function and add ```timeFrameLabel``` function which is used for the aria-label and colors 

## Related Issue

Closes #12 

## Acceptance Criteria

- Los artículos de la lista se muestran visualmente distintos (por ejemplo, con un color de fondo diferente el artículo en la lista) de acuerdo con la rapidez con la que se espera que el artículo se compre de nuevo: Pronto, Más o menos pronto, No pronto, Inactivo.
- Los artículos deben ordenarse por el número estimado de días hasta la próxima compra.
- Los artículos con el mismo número de días estimados hasta la próxima compra deben ordenarse alfabéticamente.
- Los artículos en los diferentes estados deben describirse claramente cuando los lee un lector de pantalla.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
| ✓ | :sparkles: New feature     |
| ✓ | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
| ✓ | :scroll: Docs              |

## Updates

### Before
![image](https://user-images.githubusercontent.com/42523734/141193183-f96421e0-c1c2-4b76-9dfe-c095363eb7a5.png)


### After
![image](https://user-images.githubusercontent.com/42523734/141193225-d37ef3fd-82d8-4819-a2aa-8811a179c864.png)



## Testing Steps / QA Criteria

- From your terminal, pull down this branch with ```git pull sr-yu-ordering-by-time```  and check that branch out with  ```git switch sr-yu-ordering-by-time```  
- npm start to launch the app.
